### PR TITLE
Work with Form.js wrapper and refactor address form

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

- Updating `Form.js` to be able drill down the inputs inside another component file
- Refactor `AddressForm`
  - Remove `withLocalize` HOC and use the passed `translate` props from the parent component
  - Adding new props `inputKeys` and `shouldSaveDraft` for `Form.js` usage
  - Updating the inputs passing `inputKeys` according to the key name and `shouldSaveDraft` for `Form.js` usage
- Updating the component that uses `AddressForm` to pass `translate` props

### Fixed Issues

$ https://github.com/Expensify/App/issues/10738

### Tests

1. Open a form that uses `AddressForm` for example `CompanyStep` form
2. Go to Workspace > Connect bank account > Connect Manually > Connect Bank Account > Company Information Form
3. Verified that the UI is not changed on **Company Address**, **City**, **State**, and **Zip Code** inputs.
4. Verified that the validation is working for the above inputs
5. Verified that the form successfully submit and the value of the above input is saved

- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. Open a form that uses `AddressForm` for example `CompanyStep` form
2. Go to Workspace > Connect bank account > Connect Manually > Connect Bank Account > Company Information Form
3. Verified that the UI is not changed on **Company Address**, **City**, **State**, and **Zip Code** inputs.
4. Verified that the validation is working for the above inputs
5. Verified that the form successfully submit and the value of the above input is saved
- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25520267/194902765-85332eae-abc7-4b5a-976d-408696ecd4eb.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/25520267/194902826-58b38384-6ebe-4766-b229-a81f24b5f0c4.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/25520267/194902845-ebc9e812-28bb-4c32-91d0-b878b5c8217d.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/194902856-272d0483-5fd4-4739-bef7-60eaab1ff6dc.mov

#### iOS

https://user-images.githubusercontent.com/25520267/194902875-caf05949-3e0e-41b9-9143-60f8a21178d0.mov

#### Android

https://user-images.githubusercontent.com/25520267/194902904-eb99e385-a877-464c-9b2e-4247ec2f6724.mov

### CODE REVIEW

File: src/pages/ReimbursementAccount/AddressForm.js [hunk 1]
```diff
@@ -34,32 +33,59 @@ const propTypes = {
...
+    /** Saves a draft of the input value when used in a form */
+    shouldSaveDraft: PropTypes.bool,
+
+    /** Returns translated string for given locale and phrase */
+    translate: PropTypes.func.isRequired,
 };
 
 const defaultProps = {
     values: {
+        street: undefined,
```
Comment (mananjadhav) [inline src/pages/ReimbursementAccount/AddressForm.js:53]: Any specific reason we're setting them as undefined and not `''` ?

File: src/pages/ReimbursementAccount/AddressForm.js [hunk 2]
```diff
@@ -14,7 +13,7 @@ const propTypes = {
     streetTranslationKey: PropTypes.string.isRequired,
 
     /** Callback fired when a field changes. Passes args as {[fieldName]: val} */
-    onFieldChange: PropTypes.func.isRequired,
+    onFieldChange: PropTypes.func,
```
Comment (mananjadhav) [inline src/pages/ReimbursementAccount/AddressForm.js:16]: We're removing `isRequired` because?

File: src/components/Form.js [hunk 3]
```diff
@@ -148,6 +148,20 @@ class Form extends React.Component {
                 });
             }
 
+            // Drill down the component that have a role as form wrapper
+            // e.g usage of AddressForm or IdentityForm
+            if (_.isFunction(child.type)) {
+                const nestedChildren = new child.type(child.props);
+
+                if (!React.isValidElement(nestedChildren) || !nestedChildren.props.children) {
```
Comment (mananjadhav) [inline src/components/Form.js:156]: Are we always sure we'll have `nestedChildren.props`? I am wondering if this needs to be replaced with `lodash.get` ?

File: src/pages/ReimbursementAccount/AddressForm.js [hunk 4]
```diff
@@ -34,32 +33,59 @@ const propTypes = {
...
+            <AddressSearch
+                inputID={props.inputKeys.street}
+                shouldSaveDraft={props.shouldSaveDraft}
+                label={props.translate(props.streetTranslationKey)}
+                containerStyles={[styles.mt4]}
+                value={props.values.street}
+                onInputChange={props.onFieldChange}
+                errorText={props.errors.street ? props.translate('bankAccount.error.addressStreet') : ''}
+                hint={props.translate('common.noPO')}
+                renamedInputKeys={props.inputKeys}
```
Comment (mananjadhav) [inline src/pages/ReimbursementAccount/AddressForm.js:81]: We're setting these empty because?

**Review by mananjadhav — COMMENTED** [review]
Few comments and questions, @mollfpr. Can you help address them?

File: src/components/Form.js (hunk 3 above)
Comment (mollfpr) [reply src/components/Form.js:156]: Good thinking! I'm not really sure but let's use `lodash.get` instead.

File: src/pages/ReimbursementAccount/AddressForm.js (hunk 4 above)
Comment (mollfpr) [reply src/pages/ReimbursementAccount/AddressForm.js:81]: Which one do you mean?

**Comment by mollfpr** [comment]
For testing the `AddressForm` works with `Form.js` I'm using `CompanyStep` and doing a small refactor.

```diff
diff --git a/src/pages/ReimbursementAccount/CompanyStep.js b/src/pages/ReimbursementAccount/CompanyStep.js
index ac911276f..a8e6ae620 100644
--- a/src/pages/ReimbursementAccount/CompanyStep.js
+++ b/src/pages/ReimbursementAccount/CompanyStep.js
@@ -27,6 +27,7 @@ import ReimbursementAccountForm from './ReimbursementAccountForm';
 import * as ReimbursementAccount from '../../libs/actions/ReimbursementAccount';
 import * as ReimbursementAccountUtils from '../../libs/ReimbursementAccountUtils';
 import * as FormActions from '../../libs/actions/FormActions';
+import Form from '../../components/Form';

 const propTypes = {
     ...withLocalizePropTypes,
@@ -188,7 +189,32 @@ class CompanyStep extends React.Component {
                     onCloseButtonPress={Navigation.dismissModal}
                 />

-                <ReimbursementAccountForm onSubmit={this.submit}>
+                <Form
+                    formID="testForm"
+                    submitButtonText="Submit"
+                    validate={(values) => {
+                        const errors = {};
+
+                        if (!ValidationUtils.isRequiredFulfilled(values.addressStreet)) {
+                            errors.addressStreet = this.props.translate('bankAccount.error.addressStreet');
+                        }
+
+                        if (!ValidationUtils.isRequiredFulfilled(values.addressCity)) {
+                            errors.addressCity = this.props.translate('bankAccount.error.addressCity');
+                        }
+
+                        if (!ValidationUtils.isRequiredFulfilled(values.addressState)) {
+                            errors.addressState = this.props.translate('bankAccount.error.addressState');
+                        }
+
+                        if (!ValidationUtils.isRequiredFulfilled(values.addressZipCode)) {
+                            errors.addressZipCode = this.props.translate('bankAccount.error.zipCode');
+                        }
+
+                        return errors;
+                    }}
+                    onSubmit={this.submit}
+                >
                     <Text>{this.props.translate('companyStep.subtitle')}</Text>
                     <TextInput
                         label={this.props.translate('companyStep.legalBusinessName')}
@@ -199,31 +225,14 @@ class CompanyStep extends React.Component {
                         errorText={this.getErrorText('companyName')}
                     />
                     <AddressForm
+                        shouldSaveDraft
                         translate={this.props.translate}
                         streetTranslationKey="common.companyAddress"
-                        values={{
-                            street: this.state.addressStreet,
-                            city: this.state.addressCity,
-                            zipCode: this.state.addressZipCode,
-                            state: this.state.addressState,
-                        }}
-                        errors={{
-                            street: this.getErrors().addressStreet,
-                            city: this.getErrors().addressCity,
-                            zipCode: this.getErrors().addressZipCode,
-                            state: this.getErrors().addressState,
-                        }}
-                        onFieldChange={(values) => {
-                            const renamedFields = {
-                                street: 'addressStreet',
-                                state: 'addressState',
-                                city: 'addressCity',
-                                zipCode: 'addressZipCode',
-                            };
-                            _.each(values, (value, inputKey) => {
-                                const renamedInputKey = lodashGet(renamedFields, inputKey, inputKey);
-                                this.clearErrorAndSetValue(renamedInputKey, value);
-                            });
+                        inputKeys={{
+                            street: 'addressStreet',
+                            city: 'addressCity',
+                            state: 'addressState',
+                            zipCode: 'addressZipCode',
                         }}
                     />
                     <TextInput
@@ -297,7 +306,7 @@ class CompanyStep extends React.Component {
                         style={[styles.mt4]}
                         errorText={this.getErrorText('hasNoConnectionToCannabis')}
                     />
-                </ReimbursementAccountForm>
+                </Form>
             </>
         );
     }
```


https://user-images.githubusercontent.com/25520267/195132732-0e69a623-8561-4ae9-9c8e-ee8e59dd3b37.mov

**Review by mananjadhav — DISMISSED** [review]
Thanks for the PR @mollfpr. Apologies it took a bit longer, as needed to ensure that current flow doesn't break as well as it should work with new Form.js.

Just to keep everyone updated, we are only refactoring the `AddressForm` to make it compatible with `Form.js`. It should be backward compatible, and the other forms will be using this refactored component such as https://github.com/Expensify/App/issues/9580


@puneetlath All yours. Can we also tag @luacmartins as he was involved in the solutioning and would be great to have another pair of eyes for the review.


🎀 👀 🎀 
C+ reviewed


- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by mananjadhav** [comment]
@mollfpr The check list is failing. Can you confirm if it is inline with the latest template?

@puneetlath I guess the checklist action isn't working fine.

**Comment by mollfpr** [comment]
@mananjadhav Thank you for the review!

I just pulled the main branch and updated the author checklist. The checklist running well now.

**Comment by mananjadhav** [comment]
Thanks @mollfpr.

@puneetlath @luacmartins I am wondering if we need to update any documents regarding Form with a note about HOC

**Comment by luacmartins** [comment]
>I am wondering if we need to update any documents regarding Form with a note about HOC

@mananjadhav @mollfpr yes, let's update FORMS.md to mention that children component's can't be wrapped in HOC

**Comment by mollfpr** [comment]
@mananjadhav @luacmartins Updated `FORMS.md`

File: contributingGuides/FORMS.md [hunk 5]
```diff
@@ -198,6 +198,38 @@ function onSubmit(values) {
 </Form>
 ```
 
+`Form.js` also works with inputs component that's nested on a component that doesn't wrap with any HOC.
```
Comment (mananjadhav) [inline contributingGuides/FORMS.md:201]: Can we simplify this statement? Even break it into two lines if required. 

`Form.js` also works with inputs from a nested component. The only exception is that the nested component shouldn't be wrapped around any HoC.

@luacmartins can also share thoughts on this one.

Comment (mollfpr) [reply contributingGuides/FORMS.md:201]: >Form.js also works with inputs from a nested component. The only exception is that the nested component shouldn't be wrapped around any HoC.

I'm onboard with your suggestion.

**Comment by luacmartins** [comment]
@mananjadhav could you please update your checklist to the latest version (found [here](https://github.com/Expensify/App/blob/ca501d965217c42ddc30f3516c1db7c10361c461/.github/PULL_REQUEST_TEMPLATE.md#L110)) and make sure that you upload videos for all platforms?

**Review by luacmartins — COMMENTED** [review]
Looks good. Just a couple of comments.

**Review by luacmartins — APPROVED** [review]
LGTM and tests well. @puneetlath all yours.

**Comment by luacmartins** [comment]
@mananjadhav can you please update your reviewer checklist to the [latest version](https://github.com/Expensify/App/blob/5369f7878c78456656151d18cf2e772a5bb99559/.github/PULL_REQUEST_TEMPLATE.md#pr-reviewer-checklist) and make sure that you add videos for all platforms please?

**Review by puneetlath — APPROVED** [review]

**Comment by puneetlath** [comment]
Approved on my end. @mananjadhav bump me for a merge once you've updated the PR reviewer checklist as requested [here](https://github.com/Expensify/App/pull/11611#issuecomment-1285880705).

**Comment by mananjadhav** [comment]
Yes I am on it. This one's on my todo today.

**Comment by mananjadhav** [comment]
@luacmartins @puneetlath Here's the checklist 

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


### Screenshots


#### Web
https://user-images.githubusercontent.com/3069065/197063996-994b16ba-9d26-4c72-bdc0-e2fb2cbc964c.mov

#### Mobile Web - Chrome
https://user-images.githubusercontent.com/3069065/197065417-2281bf88-7e8d-42b8-a433-9ba55b7e4942.mov

#### Mobile Web - Safari
https://user-images.githubusercontent.com/3069065/197065462-70f00f0e-9d5f-43b6-b4c8-056e5f07b4ad.mov

#### Desktop
https://user-images.githubusercontent.com/3069065/197064462-5702cb73-3cb9-4da8-b1ba-7b7ce1a3ec07.mov

#### iOS
https://user-images.githubusercontent.com/3069065/197066185-901ad3e4-7a89-4802-929d-27ab4d89e283.mov

#### Android
WIP

**Comment by luacmartins** [comment]
It's passing now. Just had to re-trigger it.

**Comment by luacmartins** [comment]
Tests passed. Removing `Emergency` label.

**Pull request merged**